src/app/domain/backup/services/uploadMedias.ts

```diff
@@ -77,6 +84,22 @@ export const uploadMedias = async (
       return t('services.backup.errors.appKilled')
     }
 
+    if (flag('flagship.backup.dedup')) {
+      const correspondingRemoteFile = getCorrespondingRemoteFile(mediaToUpload)
+
+      if (correspondingRemoteFile) {
+        await setMediaAsBackupedBecauseDeduplicated(
+          client,
+          mediaToUpload,
+          correspondingRemoteFile
+        )
+
+        void onProgress(await getBackupInfo(client))
```

Comment:
Not to be handled in this PR but maybe one possible perf improvement would be to keep an in-memory cache of `getBackupInfo()`?

The idea would be:
- `setBackupInfo` always saves backupInfo in the asyncStorage AND in an in-memory variable
- `getBackupInfo` returns the in-memory variable if defined, else in reads the asyncStorage

I'm not sure how much perf improvement we would get here as this would only save time when reading the state but would change nothing when saving it. Do you have some stats on read vs write perfs?

Reply:
There are some stats here with 20K pictures on Android https://github.com/cozy/cozy-flagship-app/pull/1097

So write are a lot slower but read are stil significant. Actually I like your idea a lot. Keep it in mind for next optimization.